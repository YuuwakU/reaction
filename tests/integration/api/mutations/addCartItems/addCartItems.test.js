import Factory from "/imports/test-utils/helpers/factory";
import TestApp from "/imports/test-utils/helpers/TestApp";
import { encodeCartOpaqueId, encodeCartItemsOpaqueIds } from "@reactioncommerce/reaction-graphql-xforms/cart";
import AddCartItemsMutation from "./AddCartItemsMutation.graphql";
import Logger from "@reactioncommerce/logger";

let testApp;
let addCartItems;
let fakeCart;
let fakeCartItem;
let shopId;
let originalError;

beforeAll(async () => {
  testApp = new TestApp();
  await testApp.start();
  shopId = await testApp.insertPrimaryShop();
  addCartItems = testApp.mutate(AddCartItemsMutation);
});
afterAll(async () => {
  await testApp.collections.Cart.deleteMany({});
  await testApp.stop();
});

describe("error messaging",() => {
  let logLevel = Logger.level();
  // jest still throws errors that are gracefully caught, so we'll up our logging level to hide errors
  // Logger.level("FATAL");
  beforeAll(async () => {
    fakeCart = Factory.Cart.makeOne({
      shopId: shopId,
      items:[]
    });
  });
  afterAll(async() => {
    Logger.level(logLevel);
  })
  test("expect fail when cartId not opaque", async () => {
    expect.assertions(1);
    try {
      let cartItems = [];
      result = await addCartItems({cartId: fakeCart._id, items: cartItems, token: null});
    } catch(error) {
      expect(error).toMatchSnapshot();
    }
  });
  test("cartId not found", async () => {
    expect.assertions(1);
    try {
      let cartItems = [];
      result = await addCartItems({cartId: encodeCartOpaqueId(fakeCart._id), items: cartItems, token: null});
    } catch(error) {
      expect(error).toMatchSnapshot();
    }
  });
  test("cartId not found with anonymous token", async () => {
    expect.assertions(1);
    try {
      let cartItems = [];
      result = await addCartItems({cartId: encodeCartOpaqueId(fakeCart._id), items: cartItems, token: "1234566"});
    } catch(error) {
      expect(error).toMatchSnapshot();
    }
  });
  test("incorrect item schema", async () =>{
    expect.assertions(1);
    try {
      let cartItems = [Factory.CartItem.makeOne()];
      console.log(encodeCartOpaqueId(fakeCart._id));
      console.log(JSON.stringify(cartItems));
      result = await addCartItems({cartId: encodeCartOpaqueId(fakeCart._id), items: cartItems, token: "1234566"});
    } catch(error) {
      expect(error).toEqual(false);
    }
  })
});

// describe('Single Product/Empty cart', async () => {
//   beforeAll(async () => {
//     fakeCartItem = Factory.Cart.makeOne({
//       shopId: shopId,
//       items:[]
//     });
//   });
//   test("add single cart item to anonymous cart", async () => {
//     // console.log(fakeCart);
//   });

//   test("add single cart item to known cart", async () => {

//   });
// });
