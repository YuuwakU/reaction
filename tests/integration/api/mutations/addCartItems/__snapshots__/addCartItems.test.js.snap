// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error messaging cartId not found 1`] = `
Array [
  Object {
    "extensions": Object {
      "code": "NOT_FOUND",
      "exception": Object {
        "details": Object {},
        "error": "not-found",
        "eventData": Object {},
        "isClientSafe": true,
        "reason": "Cart not found",
      },
    },
    "locations": Array [
      Object {
        "column": 3,
        "line": 2,
      },
    ],
    "message": "Cart not found",
    "path": Array [
      "addCartItems",
    ],
  },
]
`;

exports[`error messaging cartId not found with anonymous token 1`] = `
Array [
  Object {
    "extensions": Object {
      "code": "NOT_FOUND",
      "exception": Object {
        "details": Object {},
        "error": "not-found",
        "eventData": Object {},
        "isClientSafe": true,
        "reason": "Cart not found",
      },
    },
    "locations": Array [
      Object {
        "column": 3,
        "line": 2,
      },
    ],
    "message": "Cart not found",
    "path": Array [
      "addCartItems",
    ],
  },
]
`;

exports[`error messaging expect fail when cartId not opaque 1`] = `
Array [
  Object {
    "extensions": Object {
      "code": "INTERNAL_SERVER_ERROR",
    },
    "locations": Array [
      Object {
        "column": 3,
        "line": 2,
      },
    ],
    "message": "ID namespace must be reaction/cart",
    "path": Array [
      "addCartItems",
    ],
  },
]
`;
